### PR TITLE
[15.0][ADD] merge_duplicate_contact

### PR DESCRIPTION
**Pull request by dsolanki-initos**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
There's already https://github.com/OCA/partner-contact/tree/14.0/partner_deduplicate_filter for this.

**Comment by dsolanki-initos** [comment]
> There's already https://github.com/OCA/partner-contact/tree/14.0/partner_deduplicate_filter for this.
Hello @pedrobaeza,
Yes, but it does not have same functionality please find the screenshot of functionality of "merge_duplicate_contact" for your reference

merge_duplicate

**Comment by pedrobaeza** [comment]
OK, as you prefer. For me the other is enough.

**Review by ikapasi-initos — APPROVED** [review]
Code review and functional test.
LGTM

**Comment by rousseldenis** [comment]
@dsolanki-initos Thanks for this.

As this flow can be critical, can you add tests ? Thanks

**Comment by dsolanki-initos** [comment]
> @dsolanki-initos Thanks for this.
> 
> As this flow can be critical, can you add tests ? Thanks

I've added test cases

**Comment by dsolanki-initos** [comment]
> @dsolanki-initos Thanks for this.
> 
> As this flow can be critical, can you add tests ? Thanks

I've added test cases and you can review it

**Comment by dsolanki-initos** [comment]
@sebalix Can you please review this PR?

**Comment by dsolanki-initos** [comment]
@moylop260 Can you please review this PR?

**Comment by dsolanki-initos** [comment]
@flotho Can you please review this PR?

**Comment by dsolanki-initos** [comment]
@Galoperd Could you please review this PR?

File: merge_duplicate_contacts/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    street_no = fields.Char()
```
Comment (rousseldenis) [inline merge_duplicate_contacts/models/res_partner.py]: Why this here ?

File: merge_duplicate_contacts/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    def name_get(self):
+        ctx = self._context.copy() or {}
```
Comment (rousseldenis) [inline merge_duplicate_contacts/models/res_partner.py:11]: Could you add a comment that explain what it does ?

File: merge_duplicate_contacts/wizard/merge_contact.py [hunk 3]
```diff
@@ -0,0 +1,935 @@
...
+from ast import literal_eval
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class Partner(models.Model):
+    _inherit = "res.partner"
```
Comment (rousseldenis) [inline merge_duplicate_contacts/wizard/merge_contact.py]: Could you put one model per file ?

The res.partner code should be moved in models

File: merge_duplicate_contacts/wizard/merge_contact.py [hunk 4]
```diff
@@ -0,0 +1,935 @@
...
+
+    def open_wizard_action(self):
+        context = {}
+        data = {
```
Comment (rousseldenis) [inline merge_duplicate_contacts/wizard/merge_contact.py]: If these are fields defaults, remove entries values that are ... default.

Just add a _prepare_...() function if one want to override it.

File: merge_duplicate_contacts/wizard/merge_contact.py [hunk 5]
```diff
@@ -0,0 +1,935 @@
...
+            ignore_occurence=False,
+        )
+        return wizard._action_new_next_screen()
+
+
+class MergePartnerLine(models.TransientModel):
+    _inherit = "base.partner.merge.line"
+    _order = "len_aggr_ids asc, min_id asc"
+
+    len_aggr_ids = fields.Integer(
```
Comment (rousseldenis) [inline merge_duplicate_contacts/wizard/merge_contact.py]: An explicit field name is better than shortcuts (even if Odoo does it).

Moreover, for a count, it's better to name it with count `aggr_ids_count`

**Review by rousseldenis — COMMENTED** [review]
@dsolanki-initos This deserves a little cleaning.

Haven't you found a better way to compute an the danger icon on each field ?

Or to display a list of values that differ ?

**Comment by dsolanki-initos** [comment]
@rousseldenis, I've done all the changes as per your feedback. Could you please review this PR?

**Pull request closed without merging**